deepneuralnetwork.py

```diff
@@ -35,12 +35,11 @@ def build(self, input_shape):
         super(SelfAttLayer, self).build(input_shape)
 
     def call(self, x, mask=None):
-        eij = K.tanh(K.dot(x, self.W))
+        eij = K.tanh(K.squeeze(K.dot(x, K.expand_dims(self.W)), axis=-1))
         ai = K.exp(eij)
-        weights = ai/K.sum(ai, axis=1).dimshuffle(0,'x')
-        weighted_input = x*weights.dimshuffle(0,1,'x')
-        self.attention = weights
```

Comment:
Haven't played with this in a long time, but it seems that you are deleting this line. Is it not being used?

Reply:
Good point thanks - it got lost while moving the code